common/utils.go

```diff
@@ -18,3 +23,83 @@ func ErrorMessages(errors []error) string {
 	}
 	return strings.Join(result, "\n")
 }
+
+func WithNetNS(ns netns.NsHandle, work func() error) error {
+	errChan := make(chan error)
+
+	go func() {
+		runtime.LockOSThread()
+		defer runtime.UnlockOSThread()
+
+		oldNs, err := netns.Get()
+		if err == nil {
+			defer oldNs.Close()
+
+			err = netns.Set(ns)
+			if err == nil {
+				defer netns.Set(oldNs)
+
+				err = work()
+			}
+		}
+
+		errChan <- err
+	}()
+
+	return <-errChan
+}
+
+type WeaveInterface struct {
```

Comment:
Looks appealing, but I don't think I can use it - the `Addrs()` function is hardwired behind the scenes to look things up in the default netns via netlink. Furthermore, `net.Interface` is a struct and not an interface, so I can't provide an alternative implementation.